multicluster/controllers/multicluster/clusterset_controller.go

```diff
@@ -17,20 +17,40 @@ limitations under the License.
 package multicluster
 
 import (
+	"antrea.io/antrea/multicluster/controllers/multicluster/common"
+	"antrea.io/antrea/multicluster/controllers/multicluster/internal"
 	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	"go.uber.org/multierr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sync"
 
+	multiclusterv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/log"
-
-	multiclusterv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
 )
 
 // ClusterSetReconciler reconciles a ClusterSet object
+// There will be one controller running in each namespace of the leader for multiple cluster-set support
+// so each controller will only be handling a single cluster set in the given namespace.
 type ClusterSetReconciler struct {
 	client.Client
-	Scheme *runtime.Scheme
+	mutex    sync.Mutex
+	Scheme   *runtime.Scheme
+	Log      logr.Logger
+	IsLeader bool
+
+	clusterSetConfig *multiclusterv1alpha1.ClusterSet
+	clusterSetID     common.ClusterSetID
+	clusterID        common.ClusterID
+
+	// These fields are only applicable on leader cluster
+	inboundLeaderMgr internal.LocalClusterManager
+
+	// These fields are only applicable on member cluster
+	remoteClusterMgr internal.RemoteClusterManager
```

Comment:
should the first letter of `inboundLeaderMgr` and `remoteClusterMgr` be upper case so they can be reused in other controllers? otherwise I didn't see how common area can be used without creating a new `RemoteClusterManager`